omniisaacgymenvs/robots/sensors/proprioceptive/imu.py

```diff
@@ -0,0 +1,141 @@
+                ) #+ self._accelerometer_bias[i]
+        
+        # transform accel/gyro from body frame to sensor optical frame
+        angular_velocity = torch.bmm(self.sensor_frame_to_optical_frame[None, :3, :3].expand(angular_velocity.shape[0], 3, 3).to(device), 
+                                     torch.bmm(
+                                         self.body_to_sensor_frame[None, :3, :3].expand(angular_velocity.shape[0], 3, 3).to(device), angular_velocity[:, :, None]
+                                         )).squeeze()
+        linear_acceleration = torch.bmm(self.sensor_frame_to_optical_frame[None, :3, :3].expand(linear_acceleration.shape[0], 3, 3).to(device), 
+                                     torch.bmm(
+                                         self.body_to_sensor_frame[None, :3, :3].expand(linear_acceleration.shape[0], 3, 3).to(device), linear_acceleration[:, :, None]
+                                         )).squeeze()
+        self._sensor_state.update(angular_velocity, -1*linear_acceleration)
+    
+    @property
+    def state(self):
+        """
+        return sensor state."""
+        return self._sensor_state
+
+if __name__ == "__main__":
+    ## comes from yaml parsed by hydra ##########
+    BODY_TO_SENSOR_FRAME = [[1, 0, 0, 0], 
+                                      [0, 1, 0, 0], 
+                                      [0, 0, 1, 0], 
+                                      [0, 0, 0, 1]]
+    SENSOR_FRAME_TO_OPTICAL_FRAME = [[-0, -1, 0, 0], 
+                                     [0, 0, -1, 0], 
+                                     [1, 0, 0, 0], 
+                                     [0, 0, 0, 1]]
+    GRAVITY_VECTOR = [0, 0, -9.81]
+    dt = 0.01
+    ACCEL_PARAM = {"noise_density": 0.004, 
+                  "random_walk": 0.006, 
+                  "bias_correlation_time": 300.0, 
+                  "turn_on_bias_sigma": 0.196
+                  }
+    GYRO_PARAM = {"noise_density": 0.0003393695767766752,
+                    "random_walk": 3.878509448876288e-05,
+                    "bias_correlation_time": 1.0e3,
+                    "turn_on_bias_sigma": 0.008726646259971648
+                    }
+    #############################################
+    imu_t = IMU_T(
+        body_to_sensor_frame=BODY_TO_SENSOR_FRAME,
+        sensor_frame_to_optical_frame=SENSOR_FRAME_TO_OPTICAL_FRAME, 
+        gravity_vector=GRAVITY_VECTOR,
+        dt=dt,
+        accel_param=Accelometer_T(**ACCEL_PARAM),
+        gyro_param=Gyroscope_T(**GYRO_PARAM),
+        )
+    imu = IMUInterface(imu_t)
+
+    while True:
+        N = 16
+        position = torch.zeros(N, 3).to(torch.float32)
+        orientation = torch.zeros(N, 4).to(torch.float32)
+        orientation[:, 0] = 1.0
+        linear_velocity = torch.zeros(N, 3).to(torch.float32)
+        angular_velocity = torch.zeros(N, 3).to(torch.float32)
+        state = State(position, orientation, linear_velocity, angular_velocity)
+        imu.update(state)
```

Comment:
Should we implement python tests?

Reply:
@elharirymatteo that could be nice so that we don't have to perform a bazillion tests when doing big fat merges....